corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -625,6 +625,13 @@ def render(self, context):
     return AddToBlockNode(nodelist, 'js-inline')
 
 
+@register.simple_tag
+def html_attr(value):
+    if not isinstance(value, basestring):
+        value = JSON(value)
+    return escape(value)
```

Comment:
This seems better, but how does one know when to JSON-parse the value in the attribute and when to just use the raw attribute value? Does the JS code always know the type of things that are stuffed into attributes (at least whenever the thing is a string)? Otherwise, it seems like the JS might try to `JSON.stringify()` an attribute value and get an error because the value that was put in the attribute was a string (and therefore was not JSON-encoded).

Ugh! this is confusing. Does any of that make sense?